cdap-api-common/src/main/java/co/cask/cdap/internal/io/SchemaTypeAdapter.java

```diff
@@ -310,6 +321,13 @@ private Schema readUnion(JsonReader reader, Map<String, Schema> knownRecords) th
    * @throws IOException When fails to encode the schema into json.
    */
   private JsonWriter write(JsonWriter writer, Schema schema, Set<String> knownRecords) throws IOException {
+    if (schema.isLogicalType()) {
```

Comment:
Better writes the logical type after the main type.

Reply:
This is because logical type also has simple type.